corehq/apps/users/models.py

```diff
@@ -3107,18 +3109,37 @@ class Meta(object):
         unique_together = ('user', 'name')
 
     def save(self, *args, **kwargs):
-        if not self.key:
-            self.key = self.generate_key()
+        if not self.plaintext_key:
+            self.plaintext_key = self._generate_key()
             if 'update_fields' in kwargs:
                 kwargs['update_fields'].append('key')
+                kwargs['update_fields'].append('encrypted_key')
 
         return super().save(*args, **kwargs)
 
-    def generate_key(self):
+    def _generate_key(self):
         # From tastypie
         new_uuid = uuid4()
         return hmac.new(new_uuid.bytes, digestmod=sha1).hexdigest()
```

Comment:
Out of scope for this PR, but it might also be worth reevaluating the suitability of this approach vs say the [secrets](https://docs.python.org/3/library/secrets.html#module-secrets) module